### PR TITLE
feat(mm2): update to last mm2 + add update_maker_order rpc call + fallback smart contract

### PR DESCRIPTION
**Pull request by Milerius**
close https://github.com/KomodoPlatform/atomicDEX-Desktop/issues/736

Configure your cfg ini as follow:

```ini
[General]
2FA=0
AutomaticUpdateOrderBot=true
AvailableLang=en, fr, tr, ru
CurrentLang=fr
CurrentTheme=Binance.json
FontMode=1
MaximumNbCoinsEnabled=50
ThemePath=/Users/milerius/Library/Application Support/AtomicDex Desktop/themes
WalletChartsCategory=3

[QTUM_KMD]
Disabled=false
Max=true
MinVolume=25
Spread=2.5
```

Select the pair `QTUM/KMD` it's will prefill your order as max=true with the defined spread and volume. (click on start swap)

Each time you go to dex page, you're order should be updated, and every 2min if you do not go to the dex page

please give a try @cipig @smk762 

@shamardy please if you see anything wrong in the implementation of update_maker_order please let me know

### CODE REVIEW

File: src/core/atomicdex/api/mm2/rpc.update.maker.order.cpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+//! Project Headers
+#include <atomicdex/api/mm2/rpc.update.maker.order.hpp>
+
+namespace mm2::api
+{
+    void
+    to_json(nlohmann::json& j, const mm2::api::update_maker_order_request& request)
+    {
+        j["uuid"]      = request.uuid;
+        j["new_price"] = request.new_price;
```
Comment (Milerius) [inline src/core/atomicdex/api/mm2/rpc.update.maker.order.cpp:17]: after discussion with @shamardy, i will turn it as an optional value

File: src/core/atomicdex/api/mm2/rpc.update.maker.order.cpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        j["max"]       = request.max;
```
Comment (Milerius) [inline src/core/atomicdex/api/mm2/rpc.update.maker.order.cpp:18]: max doesn't need to be optional because it's a boolean

**Comment by smk762** [comment]
cfg.ini is global - can this be implemented "per wallet"?

**Comment by Milerius** [comment]
> cfg.ini is global - can this be implemented "per wallet"?

Unfortunately this is a bit complicated for the moment but the good thing is that it's not intrusive, this will not place orders for you, you need a manual action.

**Comment by smk762** [comment]
cool - I see it updating order form, but...
image
 Shouldn't that be +3%, not -3%?

**Comment by Milerius** [comment]
No it's should 3% you try to retrieve LTC 3% less expensive, but you try to
sell ZEC 3% more expensive

**Review by smk762 — APPROVED** [review]
Working as expected (misunderstanding aside)

**Comment by Milerius** [comment]
> Working as expected (misunderstanding aside)

i'm waiting for mm2 to be officialy updated to update mm2 here and merge this pr

**Comment by smk762** [comment]
no issues with new mm2 found with general smoke test.
auto-order update value works, but date / time value in "orders" tab does not update.
image

**Review by smk762 — APPROVED** [review]

**Comment by smk762** [comment]
ETH - BCH (`AtomicDex Desktop 0.4.3-beta / 2.1.3635_dev_a839d2f38_Linux_Release` vs `mpm / a839d2f38`): 
two new - `used new contract 0x24abe4c71fc658c91313b6552cd40cd808b3ea80`
BNB - ETH-BEP20 (`AtomicDex Desktop 0.4.3-beta / 2.1.3635_dev_a839d2f38_Linux_Release` vs `mpm / a839d2f38`): 
two new - `used new contract 0xedc5b89fe1f0382f9e4316069971d90a0951db31`
Need to move some funds around to test other cases

**Review by tonymorony — APPROVED** [review]
lgtm

**Pull request merged**